src/java.base/share/classes/java/lang/reflect/Method.java

```diff
@@ -567,12 +567,40 @@ public Object invoke(Object obj, Object... args)
     }
 
     /**
-     * Returns {@code true} if this method is a bridge
-     * method; returns {@code false} otherwise.
+     * {@return {@code true} if this method is a bridge
+     * method; returns {@code false} otherwise}
+     *
+     * @apiNote
+     * A bridge method is a {@linkplain isSynthetic synthetic} method created by a Java
+     * compiler alongside a method originating from the source
+     * code. One example use of a bridge method is as technique for a
+     * Java compiler to support <i>covariant overrides</i>, where a
+     * subclass overrides a method and gives the new method a more
+     * specific return type than the method in the superclass. A
+     * common case where covariant overrides are used is for a {@link
+     * java.lang.Cloneable Cloneable} class where the {@code clone}
+     * method inherited from {@code java.lang.Object} is overridden
+     * and declared to return the type of the class; for example
+     * {@link java.util.EnumSet#clone() EnumSet.clone()} returns
+     * {@code EnumSet<E>} rather than {@link java.lang.Object#clone()
+     * Object}. If this technique was being used, the resulting class
+     * file for {@code EnumSet} would have two {@code clone} methods,
+     * one returning {@code EnumSet<E>} and the second a bridge method
+     * returning {@code Object}; the body of the {@code clone} bridge
+     * method calls its non-bridge counterpart and returns its
+     * result. (While the Java language specification forbids a class
+     * declaring two methods with the same parameter types but a
+     * different return type, the virtual machine does not.)
+     *
+     * <p>Bridge methods may also be used by Java compiler in other
+     * circumstances to span across difference in Java Language
+     * semantics and JVM semantics.
```

Comment:
If you decide to put less detail here, you could start with this statement. I think the main point is that there are some semantic gaps between methods in the Java language and in the JVM; bridge methods are necessary to "span" this gap. You might simply list some examples without explaining them fully.

Would this be "used by **a** Java compiler" or "used by **the** Java compiler?

Reply:
Imo "created" should be better. Is there any case where a java compiler accepts bridge methods generated?